src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,337 @@
+    
+    #NOTE: Thw while loop in here will not be needed once you fix the "FileNameGeneration" problem.
+    def start_acquisition_and_wait(self) -> None:
+        """
+        Starts the acquisition process and waits until it is completed before carrying on.
+        """
+        acquisition_time_seconds = self.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime) / 1000.0
+        num_frames = self.get(lf.AddIns.ExperimentSettings.AcquisitionFramesToStore)
+        self.experiment.Acquire()
+
+        sleep(num_frames * acquisition_time_seconds) 
+        
+        while self.experiment.IsRunning:
+            sleep(0.1)  # checking if the experiment is still running
+
+    def process_acquired_data(self) -> np.ndarray:
+        """
+        Processes the most recently acquired data and returns it as a numpy array.
+        """
+        last_file = self._application.FileManager.GetRecentlyAcquiredFileNames()[0]
+        image_dataset = self._application.FileManager.OpenFile(last_file, FileAccess.Read)
+
+        if image_dataset.Regions.Length == 1:
+            if image_dataset.Frames == 1:
+                return self._process_single_frame(image_dataset)
+            else:
+                return self._process_multiple_frames(image_dataset)
+        else:
+            raise QT3Error(f"LightField FileManager OpenFile error. Unsupported value for Regions.Length: {image_dataset.Regions.Length}.Should be == 1")
+
+    def _process_single_frame(self, image_dataset: Any) -> np.ndarray:
+        """
+            **Single Frame**:
+            - Data is returned as a 2D array representing raw counts from each pixel.
+            - A vertical section (spanning 400 pixels) represents a range for wavelength averaging.
+        """
+        frame = image_dataset.GetFrame(0, 0)
+        return np.reshape(np.fromiter(frame.GetData(), dtype='uint16'), [frame.Width, frame.Height], order='F')
+
+    def _process_multiple_frames(self, image_dataset: Any) -> np.ndarray:
+        """
+            **Multiple Frames**:
+            - Data from successive exposures is added as a new dimension, then returns a 3D array.
+            - Averaging across frames can be done by summing over this additional dimension.
+        """
+        data = np.array([])
+        for i in range(image_dataset.Frames):
+            frame = image_dataset.GetFrame(0, i)
+            new_frame = np.reshape(np.fromiter(frame.GetData(), dtype='uint16'), [frame.Width, frame.Height], order='F')
+            data = np.dstack((data, new_frame)) if data.size else new_frame
+        return data
+
+    #NOTE: May not need to call the 'start_acquisition_and_wait' method if you fix the 'FileNameGeneration' issue
+    def acquire(self) -> np.ndarray:
+        """
+        Acquires image data from the spectrometer.
+        """
+        self.file_setup()
+        self.start_acquisition_and_wait()
+        return self.process_acquired_data()
```

Comment:
I suggest making these methods for internal use only (add a _ before their names), so that the user thinks twice before they use them independently.